### PR TITLE
feat: add admin/sales page

### LINKED ISSUE #154: Adding admin/sales route(UI only).
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

A admin/sales route.

adminsales

There are also subroutes. Will be dealt as separate issues.

### PR DESCRIPTION
**Pull request by sumedh123**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

Fixes #154 
Here's the deployment link : 
https://oefrontend2017.herokuapp.com

### CODE REVIEW

**Comment by CosmicCoder96** [comment]
@sumedh123 Please don't tick `The acceptance, integration, unit tests and linter pass locally with my changes` unless they do, the build is failing please check

File: app/templates/admin/sales.hbs [hunk 1]
```diff
@@ -1 +1,18 @@
-{{outlet}}
+<div class="ui grid">
+  <div class="row">
+    <div class="{{if device.isMobile 'sixteen' 'three'}} wide column">
```
Comment (geekyd) [inline app/templates/admin/sales.hbs]: Please replace this by tabbed-navigation.

**Comment by sumedh123** [comment]
@geekyd Please review

File: app/routes/admin/sales.js [hunk 2]
```diff
@@ -3,4 +3,5 @@ import Ember from 'ember';
 const { Route } = Ember;
 
 export default Route.extend({
+
```
Comment (niranjan94) [inline app/routes/admin/sales.js]: titleToken missing

Comment (niranjan94) [inline app/routes/admin/sales.js]: add title "Sales"

File: app/routes/admin/sales/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+export default Route.extend({
```
Comment (niranjan94) [inline app/routes/admin/sales/index.js]: Add titleToken title 'Overview'

File: app/templates/admin/sales/index.hbs [hunk 4]
```diff
@@ -0,0 +1,76 @@
+{{#if device.isMobile}}
+  <br><br>
+{{/if}}
+<div class="ui stackable grid">
+  <div class="row">
+    <h1 class="ui header six wide column">{{t 'Tickets Summary'}}</h1>
+  </div>
+  <div class="row">
+    <div class="eight wide column {{if device.isMobile 'center' 'left'}} aligned">
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Remove this. Will be handled by ember-model-table

File: app/templates/admin/sales/index.hbs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          <div class="item" data-value="">10</div>
+          <div class="item" data-value="">20</div>
+          <div class="item" data-value="">50</div>
+          <div class="item" data-value="">100</div>
+        </div>
+      {{/ui-dropdown}}
+      &nbsp;
+      <span>{{t 'entries'}}</span>
+    </div>
+    <div class="{{if device.isMobile 'eight wide column center' 'three wide column right'}} aligned">
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Remove this. Will be handled by ember-model-table

Comment (sumedh123) [inline app/templates/admin/sales/index.hbs]: @niranjan94 Shall I remove the "Showing all records" dropdown also?

Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Yes

**Comment by sumedh123** [comment]
@niranjan94 Anything else needed here?

File: app/templates/admin/sales.hbs (hunk 1 above)
Comment (niranjan94) [inline app/templates/admin/sales.hbs]: Looks like this isn't done yet.

Comment (sumedh123) [inline app/templates/admin/sales.hbs]: @niranjan94 I have added tabbed navigation with the above classes. Replacing the above classes with tabbed navigation results in bad UI. I have made the changes.

Comment (niranjan94) [inline app/templates/admin/sales.hbs]: Please see how the tabs are implemented in `events.hbs`. 
We need it like that (using the component) to make sure the tab layout changes to a different style on mobile

File: app/templates/admin/sales.hbs [hunk 6]
```diff
@@ -1 +1,18 @@
-{{outlet}}
+<div class="ui grid">
+  <div class="row">
+    <div class="{{if device.isMobile 'sixteen' 'three'}} wide column tabbed-navigation">
```
Comment (niranjan94) [inline app/templates/admin/sales.hbs]: Remove the `tabbed-navigation` class. Doesn't make sense. 
And wrap the items below inside a `tabbed-navigation` component. Pass parameter `isVertical=true`

File: app/templates/admin/sales.hbs [hunk 7]
```diff
@@ -1 +1,32 @@
-{{outlet}}
+<div class="ui grid">
+  <div class="row">
+    <div class="{{if device.isMobile 'sixteen' 'three'}} wide column">
```
Comment (niranjan94) [inline app/templates/admin/sales.hbs]: Don't. Just put `three` and add `stackable` to the grid

File: app/templates/admin/sales.hbs [hunk 8]
```diff
@@ -1 +1,32 @@
...
+        {{/link-to}}
+        {{#link-to 'admin.sales.fees' class="item"}}
+          {{t 'Fees'}}
+        {{/link-to}}
+        {{#link-to 'admin.sales.status' class="item"}}
+          {{t 'Invoices'}}
+        {{/link-to}}
+      {{/tabbed-navigation}}
+    </div>
+    <div class="{{if device.isMobile 'sixteen' 'thirteen'}} wide column">
```
Comment (niranjan94) [inline app/templates/admin/sales.hbs]: Don't. Just put `thirteen` and add `stackable` to the grid

**Comment by sumedh123** [comment]
@niranjan94 Done.

File: app/templates/admin/sales/index.hbs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  <div class="row">
+    <h1 class="ui header six wide column">{{t 'Tickets Summary'}}</h1>
+  </div>
+  <div class="row">
+    <table class="ui stackable structured celled table">
+      <thead>
+        <tr>
+          <th rowspan="2" class="left aligned">{{t 'Events'}}</th>
+          <th rowspan="2" class="left aligned">{{t 'Event Date'}}</th>
+
```
Comment (geekyd) [inline app/templates/admin/sales/index.hbs]: Remove extra line from here.

**Comment by sumedh123** [comment]
@niranjan94 Please review.

**Comment by sumedh123** [comment]
@niranjan94 Please review

File: app/templates/admin/sales.hbs [hunk 10]
```diff
@@ -1 +1,32 @@
...
+        {{#link-to 'admin.sales.index'  class="item"}}
+          {{t 'By Events'}}
+        {{/link-to}}
+        {{#link-to 'admin.sales.organizers' class="item"}}
+          {{t 'By Organizer'}}
+        {{/link-to}}
+        {{#link-to 'admin.sales.locations' class="item"}}
+          {{t 'By Location'}}
+        {{/link-to}}
+        {{#link-to 'admin.sales.marketer' class="item"}}
```
Comment (niranjan94) [inline app/templates/admin/sales.hbs]: Single quotes for handlebar helpers please ... 
Like I said multiple times, please see https://github.com/fossasia/open-event-frontend/blob/development/docs/style-guide.md and follow the same ...

**Comment by sumedh123** [comment]
@niranjan94 Corrected the quotes. Please review. If you want the deployment link, I can push to heroku this branch once my other PR gets merged.

**Comment by geekyd** [comment]
@sumedh123 Please have at least 2 heroku deployed apps, for such cases. Thanks!

File: app/routes/admin/sales/locations.js [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
+});
```
Comment (geekyd) [inline app/routes/admin/sales/locations.js]: Add titleToken title 'Locations'

File: app/routes/admin/sales/marketer.js [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
+});
```
Comment (geekyd) [inline app/routes/admin/sales/marketer.js]: Add titleToken title 'Marketer'

File: app/routes/admin/sales/organizers.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
+});
```
Comment (geekyd) [inline app/routes/admin/sales/organizers.js]: Add titleToken title 'Organizer'

File: app/routes/admin/sales/status.js [hunk 14]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
+});
```
Comment (geekyd) [inline app/routes/admin/sales/status.js]: Add titleToken title 'Status'

Comment (sumedh123) [inline app/routes/admin/sales/status.js]: Done.

**Comment by geekyd** [comment]
@sumedh123  Please provide deployment link.

**Comment by sumedh123** [comment]
@geekyd https://oefrontend2017.herokuapp.com

File: app/routes/admin/sales/index.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+  titleToken() {
+    return this.i18n.t('Overview');
+  },
+
+  model() {
+    return [{
+      events           : 'Sample Event',
+      eventDate        : '2017-06-09',
```
Comment (geekyd) [inline app/routes/admin/sales/index.js]: This will be a JS date object.

File: app/templates/admin/sales/index.hbs [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+          <th class="right aligned">{{t 'Sales'}}</th>
+          <th class="right aligned">{{t 'Tickets'}}</th>
+          <th class="right aligned">{{t 'Sales'}}</th>
+        </tr>
+      </thead>
+      <tbody>
+        {{#each model as |entry|}}
+          <tr>
+            <td>{{entry.events}}</td>
+            <td>{{entry.eventDate}}</td>
```
Comment (geekyd) [inline app/templates/admin/sales/index.hbs]: Use moment helper here for JS date object.

**Comment by sumedh123** [comment]
@geekyd Please review. https://oefrontend2017.herokuapp.com

File: app/templates/admin/sales/index.hbs [hunk 17]
```diff
@@ -0,0 +1,40 @@
+<div class="ui stackable grid">
+  <div class="row">
+    <h1 class="ui header six wide column">{{t 'Tickets Summary'}}</h1>
+  </div>
+  <div class="row">
+    <table class="ui stackable structured celled table">
+      <thead>
+        <tr>
+          <th rowspan="2" class="left aligned">{{t 'Events'}}</th>
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: It is left aligned by default. Class not needed.

**Comment by sumedh123** [comment]
@niranjan94 Please review.

**Comment by sumedh123** [comment]
@niranjan94 is there anything needed here?

File: app/templates/admin/sales/index.hbs [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+          <th class="right aligned">{{t 'Sales'}}</th>
+        </tr>
+      </thead>
+      <tbody>
+        {{#each model as |entry|}}
+          <tr>
+            <td>{{entry.events}}</td>
+            <td>{{moment-format entry.eventDate 'ddd, MMM DD \at HH:mm A'}}</td>
+            <td class="right aligned">{{entry.completedTickets}}</td>
+            <td class="right aligned">{{entry.completedSales}}</td>
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Use `number-format` helper

File: app/templates/admin/sales/index.hbs [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+            <td class="right aligned">{{entry.placedTickets}}</td>
+            <td class="right aligned">{{entry.placedSales}}</td>
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Use `number-format` helper

File: app/templates/admin/sales/index.hbs [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+            <td class="right aligned">{{entry.pendingTickets}}</td>
+            <td class="right aligned">{{entry.pendingSales}}</td>
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Use `number-format` helper

File: app/templates/admin/sales/index.hbs [hunk 21]
```diff
@@ -0,0 +1,40 @@
+<div class="ui stackable grid">
+  <div class="row">
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Every row should have a column inside it with the content.  Please read up on Semantic UI's grid layout.

File: app/templates/admin/sales/index.hbs [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+    <h1 class="ui header six wide column">{{t 'Tickets Summary'}}</h1>
+  </div>
+  <div class="row">
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: Every row should have a column inside it with the content.  Please read up on Semantic UI's grid layout.

**Comment by sumedh123** [comment]
@niranjan94 Please review. https://oefrontend2017.herokuapp.com

File: app/templates/admin/sales/index.hbs [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+            <th class="right aligned">{{t 'Tickets'}}</th>
+            <th class="right aligned">{{t 'Sales'}}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {{#each model as |entry|}}
+            <tr>
+              <td>{{entry.events}}</td>
+              <td>{{moment-format entry.eventDate 'ddd, MMM DD \at HH:mm A'}}</td>
+              <td class="right aligned">{{number-format entry.completedTickets}}</td>
```
Comment (niranjan94) [inline app/templates/admin/sales/index.hbs]: We don't want a decimal for the number of tickets sold. We won't be selling `1.23` tickets would we ? Please think about the business logic too. We need number-format only for cost values. In this case all the sales entries.

**Pull request merged**